src/main/java/com/fakng/fakngagrgtr/parser/apple/AppleParser.java

```diff
@@ -0,0 +1,131 @@
+package com.fakng.fakngagrgtr.parser.apple;
+
+import com.fakng.fakngagrgtr.parser.HtmlParser;
+import com.fakng.fakngagrgtr.parser.LocationProcessor;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.location.Location;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlScript;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class AppleParser extends HtmlParser {
+
+    private static final String URL_FOR_VACANCY = "https://jobs.apple.com/en-us/details/";
+    private static final String ELEMENT_NAME_WITH_PAGE_COUNT = "frmPagination";
+    private static final String ELEMENT_TAG_WITH_PAGE_COUNT = "span";
+    private static final String PATH_TO_VACANCIES = "searchResults";
+    private static final String PATH_TO_FIELD_WITH_JSON_BODY = ".//script";
+    private static final JSONParser jsonParser = new JSONParser(JSONParser.MODE_JSON_SIMPLE);
+    private static final ObjectMapper mapper = new ObjectMapper();
```

Comment:
Spring has this bean ready-to-use
Please, inject it in your component. There's no need to create it with "new" operator.